app/models/language_skill.rb

```diff
@@ -1,6 +1,13 @@
 class LanguageSkill < ApplicationRecord
   belongs_to :languageable, polymorphic: true, optional: true
 
+  scope :native, -> { where(level: 'native_speaker') }
+  scope :native_languages, lambda {
+    return native if native.size <= 1
```

Comment:
I'd merge `native` and `native_languages`. Is there a specific reason the two are separate?

Alternatively I'd convert the `native_languages` to an order scope (then you could chain them like `LanguageSkill.native.ordered_by_language`).